pika/connection.py

```diff
@@ -612,9 +614,12 @@ def __init__(self,  # pylint: disable=R0913,R0914,R0912
         :param pika.credentials.Credentials credentials: auth credentials
         :param int channel_max: Maximum number of channels to allow
         :param int frame_max: The maximum byte size for an AMQP frame
-        :param int heartbeat: Heartbeat timeout. Max between this value
-            and server's proposal will be used as the heartbeat timeout. Use 0
-            to deactivate heartbeats and None to accept server's proposal.
+        :param int|callable heartbeat: Heartbeat timeout. If set and is
```

Comment:
Please add a test for passing callable heartbeat using `ConnectionParameters` class to the appropriate section in `test/unit/connection_parameters_tests.py`

Reply:
done